### PR TITLE
drivers: mcux: Convert MCUX drivers to new DT_INST macros

### PR DESCRIPTION
**Pull request by galak**
Convert older DT_INST_ macro use in MCUX drivers to the new
include/devicetree.h DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: drivers/adc/adc_mcux_adc12.c [hunk 1]
```diff
@@ -252,26 +253,26 @@ static const struct adc_driver_api mcux_adc12_driver_api = {
 #define TO_ADC12_CLOCK_SRC(val) _DO_CONCAT(kADC12_ClockSourceAlt, val)
 #define TO_ADC12_CLOCK_DIV(val) _DO_CONCAT(kADC12_ClockDivider, val)
 
-#if DT_INST_0_NXP_KINETIS_ADC12
+#if DT_HAS_NODE(DT_INST(0, nxp_kinetis_adc12))
```
Comment (henrikbrixandersen) [inline drivers/adc/adc_mcux_adc12.c]: Can we use `DT_DRV_COMPAT` definition here?

File: drivers/clock_control/clock_control_mcux_pcc.c [hunk 2]
```diff
@@ -65,12 +67,12 @@ static const struct clock_control_driver_api mcux_pcc_api = {
 	.get_rate = mcux_pcc_get_rate,
 };
 
-#if defined(DT_INST_0_NXP_KINETIS_PCC_LABEL)
+#if DT_INST_NODE_HAS_PROP(0, label)
```
Comment (henrikbrixandersen) [inline drivers/clock_control/clock_control_mcux_pcc.c]: This should likely be `#if DT_HAS_NODE(DT_INST(0, nxp_kinetis_pcc))`.

File: drivers/watchdog/wdt_cmsdk_apb.c [hunk 3]
```diff
@@ -4,6 +4,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#define DT_DRV_COMPAT arm_cmsdk_watchdog
```
Comment (henrikbrixandersen) [inline drivers/watchdog/wdt_cmsdk_apb.c]: Did these changes sneak in by mistake? They are not MCUX specific.

File: drivers/adc/adc_mcux_adc12.c (hunk 1 above)
Comment (galak) [reply drivers/adc/adc_mcux_adc12.c]: Introduced `DT_HAS_DRV_INST`, now its `#if DT_HAS_DRV_INST(0)`

File: drivers/watchdog/wdt_cmsdk_apb.c (hunk 3 above)
Comment (galak) [reply drivers/watchdog/wdt_cmsdk_apb.c]: Yeah, will remove.

File: drivers/clock_control/clock_control_mcux_pcc.c (hunk 2 above)
Comment (galak) [reply drivers/clock_control/clock_control_mcux_pcc.c]: why?  do we different compats for PCC?

Comment (galak) [reply drivers/clock_control/clock_control_mcux_pcc.c]: ah, get the point now.  Will change it over from testing `label`

**Review by henrikbrixandersen — APPROVED** [review]
LGTM, thanks.

**Review by mbolivar-nordic — APPROVED** [review]
Looks sane. I just skimmed these.

**Review by MaureenHelm — APPROVED** [review]

**Review by vanwinkeljan — APPROVED** [review]
+1 for display related changes

**Pull request merged**